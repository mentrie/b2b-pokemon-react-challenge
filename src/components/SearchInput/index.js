import React from 'react';

const SearchInput = ({ value, onChange }) => {
  return (
    <input
      type="search"
      placeholder="Search for PokÃ©mon here"
      value={value}
      onChange={onChange}
      className="input__search"
    />
  );
};

export default SearchInput;
