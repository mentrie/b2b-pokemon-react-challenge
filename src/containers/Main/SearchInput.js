import React from 'react';

const SearchInput = ({ value, onChange }) => {
  return (
    <input
      type="text"
      placeholder="Search for PokÃ©mon here"
      value={value}
      onChange={onChange}
    />
  );
};

export default SearchInput;
