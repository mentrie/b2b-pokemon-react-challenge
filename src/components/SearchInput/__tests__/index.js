import React from 'react';
import SearchInput from '../index';

test('should render component properly', () => {
  const component = shallow(<SearchInput />);
  expect(component).toMatchSnapshot();
});

test('should update the value of the input', () => {
  const onChangeMock = jest.fn();

  const component = mount(<SearchInput onChange={onChangeMock} value="test" />);
  expect(component.find('input').length).toBe(1);
  component.find('input').simulate('change');
  expect(onChangeMock).toHaveBeenCalled();
});
