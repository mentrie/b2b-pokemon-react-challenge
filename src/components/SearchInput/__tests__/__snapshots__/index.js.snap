// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component properly 1`] = `
<input
  className="input__search"
  placeholder="Search for PokÃ©mon here"
  type="search"
/>
`;
